### PR TITLE
Searchbar

### PR DESCRIPTION
**Pull request by n1lsw**
# 8 - I created a searchbar component including a story

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export function Button() {
+  return (
+    <div className="flex self-center">
+      <button
+        type="submit"
+        className="rounded-md h-[48px] w-[335px] bg-[#FFB43A]"
```
Comment (NoHara42) [inline src/components/Button.tsx:6]: don't overuse tailwind square bracket notation, h-12, and a bg-someCustomColorName should be used here.

w-[335px] is fine if you don't reuse that width

File: src/components/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { redirect, useNavigate } from "react-router-dom";
+// import { BsSearch } from "<react-icons />Bs;
+
+export function SearchBar() {
+  const navigate = useNavigate();
+  const submitfunction = async (event: { preventDefault: () => void }) => {
+    event.preventDefault();
+
+    // const formData = new FormData(form);
+    // const fromEntries = formData.entries();
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:11]: don't commit comment blocks

File: src/components/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <div className="">
+      <div className="flex items-center justify-screen gap-5 flex-col justify-center h-screen bg-gray-800">
+        <form method="GET">
+          <div className="relative focus-within:text-gray-400">
+            <span className="absolute inset-y-0 left-0 flex items-center pl-2">
+              <button
+                type="submit"
+                className="p-1 focus:outline-none focus:shadow-outline text-[#FFFFFF] opacity-40"
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:36]: why not use -white instead of -[#FFFFFF]

File: src/components/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                  className="w-6 h-6"
+                >
+                  <path d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path>
+                </svg>
+              </button>
+            </span>
+            <input
+              type="search"
+              name="q"
+              className="py-2 text-sm text-[#FFFFFF] rounded-full bg-[#363740] h-[48px] w-[335px] hover:border-2 hover:border-[#FFFFFF] hover:border-opacity-40 pl-10 focus:outline-none"
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:54]: same here, think about extracting similar sets of utility classes to custom css classes

**Comment by n1lsw** [comment]
Please ignore the comment block in the beginning. It is part of the code, but I needed to comment it to have storybook to run

File: package.json [hunk 5]
```diff
@@ -21,7 +21,9 @@
     "react": "^18.2.0",
     "react-barcode": "^1.4.6",
     "react-dom": "^18.2.0",
+    "react-icons": "^4.7.1",
     "react-router-dom": "^6.6.1",
+    "serve": "^14.1.2",
```
Comment (NoHara42) [inline package.json:26]: why is serve a dependency here

File: src/components/Movie.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import axios from "axios";
```
Comment (NoHara42) [inline src/components/Movie.tsx:1]: should be part of #23, not the searchbar issue PR

File: src/components/Test.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import { API } from "./API";
```
Comment (NoHara42) [inline src/components/Test.tsx:1]: do not commit test components

**Review by NoHara42 — CHANGES_REQUESTED** [review]
please address all my comments before sending a PR back to review.
remove -lock.json files and replace .svgs in assets with heroicon in-file imports

File: package.json (hunk 5 above)
Comment (n1lsw) [reply package.json:26]: deleted line 26

File: src/components/Movie.tsx (hunk 6 above)
Comment (n1lsw) [reply src/components/Movie.tsx:1]: removed it

File: src/components/Test.tsx (hunk 7 above)
Comment (n1lsw) [reply src/components/Test.tsx:1]: deleted it

File: package.json [hunk 8]
```diff
@@ -21,6 +21,7 @@
     "react": "^18.2.0",
     "react-barcode": "^1.4.6",
     "react-dom": "^18.2.0",
+    "react-icons": "^4.7.1",
```
Comment (NoHara42) [inline package.json:24]: where is react-icons used?

File: package.json [hunk 9]
```diff
@@ -30,7 +31,7 @@
     "@storybook/addon-interactions": "^7.0.0-beta.19",
     "@storybook/addon-links": "^7.0.0-beta.19",
     "@storybook/blocks": "^7.0.0-beta.19",
-    "@storybook/react": "^7.0.0-beta.19",
+    "@storybook/react": "7.0.0-beta.19",
```
Comment (NoHara42) [inline package.json:34]: not necessary to fix the version number

**Comment by NoHara42** [comment]
@n1lsw please see all my comments above and address them individually before sending it back for review

https://github.com/devhausleipzig/camp-8-midterm-group-1/pull/19#pullrequestreview-1243713325

remove all lock files (package-lock.json, pnpm-lock.yaml), and address other comments please

**Review by NoHara42 — APPROVED** [review]
LGTM

**Pull request merged**